### DISCUSSION TITLE
Why does Popup return object?

### DISCUSSION THREAD

**Discussion by marekm294** [CommunityToolkit/Maui#1148] [General]
I just wonder why the ShowPopupAsync method returns an object?. In XamarinForms the Popup was generic. In our MAUI application, we added extension methods that automatically cast object?, so it's a bit more readable in our VMs, but I think it would be nice to have generic popups and not have to return object?. It's possible this has been discussed before, but I haven't found it, so please redirect me if so.

**Comment by TheCodeTraveler** [comment]
Would you like to put together an API Design and submit a Proposal?

You can find more information on submitting a new feature here: https://github.com/CommunityToolkit/Maui#submitting-a-new-feature

**Reply by marekm294** [reply]
I hope you mean API for generic popup and not dummy casting extension methods :). Yeah, I could try to find time next week. I assume it should be same or simmilar to API in old Xamarin.Forms toolkit. I also think it should be compatible with current version right?

**Comment by pictos** [comment]
@marekm294 that was because of a limitation on Handlers and generic return types... There was no easy and/or general solution. If I recall the conversation that I had with the team at that time, they will look at how to improve that but isn't a very urgent task. So in order to deliver the Popup I choose to use `object` as the return type... 

I would say that you can create your generic version of that, but you will need to add a new handler for each generic type (that was the reason I choose to not implement it in a generic way)

**Comment by CliffAgius** [comment]
Closed and answered
